lib/server.js

```diff
@@ -100,6 +112,49 @@ Server.prototype._initializeOptions = function(options) {
   this.wsdl.options.attributesKey = options.attributesKey || 'attributes';
 };
 
+Server.prototype._processRequestXml = function(req, res, xml){
+  var self = this;
+  var result;
+  var error;
+  try {
+    if (typeof self.log === 'function') {
+      self.log("received", xml);
+    }
+    self._process(xml, req, function(result, statusCode) {
+      if(statusCode) {
+        res.statusCode = statusCode;
+      }
+      res.write(result);
+      res.end();
+      if (typeof self.log === 'function') {
+        self.log("replied", result);
+      }
+    });
+  }
+  catch (err) {
```

Comment:
this snippet was taken from previous code, so, I wanted to leave it as it was to keep reviews simpler. But, will correct and squash the commits.